### PR TITLE
first pass from product backend

### PR DESCRIPTION
**Pull request by jtprince**
I'm still not sure on one or two of these, but this is my first pass changes to this.  [btw, I love working with version control and I love working in markdown]

### CODE REVIEW

File: Product--Add_To_Inventory.md [hunk 1]
```diff
@@ -1,20 +1,13 @@
...
-
-### 
-
-??????????
-
+### URLs
 
+POST /products/inventory-lists/<inventory-list-uuid>/add-items
+POST /products/inventory-lists/<inventory-list-uuid>/add-skus
+POST /products/inventory-lists/<inventory-list-uuid>/add-skus-by-search
```
Comment (Gidgidonihah) [inline Product--Add_To_Inventory.md]: wouldn't the search be add items by search? Or would it truly be skus?

File: Product--READ-InventoryList-SubCatalog.md [hunk 2]
```diff
@@ -18,23 +13,24 @@ empty body
...
       },
-      "date_last_modified": <datestring>,
-      "last_modified_user": {
+      "datetime_modified": <datestring>,
+      "last_modified_by_user": {
         "uuid": <uuid>,
         "name": <string>,
       },
       "num_skus": <int>,
+      ?? What sku information is wanted / needed ??
```
Comment (Gidgidonihah) [inline Product--READ-InventoryList-SubCatalog.md]: none sku information. this list is it. Oh except we need to add:
```
images: [
// primary image of last for items added
]
```

File: Product--SKU_autocomplete.md [hunk 3]
```diff
@@ -14,6 +12,11 @@
 
 ### Response
 
+???? WHAT is the minimum amount that you actually need from this request??
+???? If you only need basic information I can return it straight from
+???? elasticsearch and this will be super fast.  If you make me return a whole
+???? bunch of stuff, then I'll have to hit the DB and it won't be as quick
+
```
Comment (Gidgidonihah) [inline Product--SKU_autocomplete.md]: name, pricing, image is probably the minimum.

File: Product--SearchCatalog-SubCatalog_and_Inventory.md [hunk 4]
```diff
@@ -12,7 +12,7 @@
...
    shipping_cost: [<uuid>, <uuid>, ...],
    shipping_origin: [<uuid>, <uuid>, ...],
    product_codes: [<uuid>, <uuid>, ...],
  },
  search_term: <string>,
  sort: <string>,
  start: <num>,
  limit: <num>,
}
+{  filters: {    supplier: [<uuid>, <uuid>, ...],    category: <uuid>,    price: { min: <num>, max: <num>},    pricing_type: [<uuid>, <uuid>, ...],    shipping_cost: [<uuid>, <uuid>, ...],    shipping_origin: [<uuid>, <uuid>, ...],    product_codes: [<uuid>, <uuid>, ...],  },  search_term: <string>,  sort: <string>,  start: <num>,  limit: <num>,}
```
Comment (Gidgidonihah) [inline Product--SearchCatalog-SubCatalog_and_Inventory.md]: Oh no, the whitespace! What actually changed here?

Comment (jtprince) [reply Product--SearchCatalog-SubCatalog_and_Inventory.md]: I just stripped out carriage returns (who uses just carriage returns? :) )  Lemme try that again and replace with newlines.

File: Product--SKU_autocomplete.md (hunk 3 above)
Comment (jtprince) [reply Product--SKU_autocomplete.md]: the first sku image?  I could easily hand you a url...

File: Product--Add_To_Inventory.md (hunk 1 above)
Comment (jtprince) [reply Product--Add_To_Inventory.md]: You told me you want inventory-lists to be sku based (which makes sense).  For consistency, sub-catalogs and inventory lists will both be sku based.  In addition, search returns skus, so it's possible that some skus match and some do not within a single item.  So, I think this is correct (adding skus by search, not adding items by search).

File: Product--SearchCatalog-SubCatalog_and_Inventory.md (hunk 4 above)
Comment (jtprince) [reply Product--SearchCatalog-SubCatalog_and_Inventory.md]: fixed.

File: Product--READ-InventoryList-SubCatalog.md (hunk 2 above)
Comment (jtprince) [reply Product--READ-InventoryList-SubCatalog.md]: how about:

`images_from_recently_added_items: [{ <image info> }, ... ]`

File: Product--SKU_autocomplete.md (hunk 3 above)
Comment (Gidgidonihah) [reply Product--SKU_autocomplete.md]: yep.

**Pull request merged**